webapps/api/backend/decorators.py

```diff
@@ -289,11 +302,25 @@ def new_function(*args, **kwargs):
             guid_list = isinstance(data_list, list) and len(data_list) > 0 and isinstance(data_list[0], basestring)
             timings['fetch'] = [time.time() - start, 'Fetching data']
 
+            # Filtering data
+            if query is not None:
+                start = time.time()
+                if guid_list is True:
+                    guids = data_list
+                    guid_list = False  # The list will be converted to a datalist
+                else:
+                    guids = data_list.guids
+                # Use the guids from the result list as a base to query inside the functions results and apply the query
+                data_list = DataList(object_type, query=query, guids=guids)
+                # Trigger the query
+                _ = data_list.guids
+                timings['querying'] = [time.time() - start, 'Querying data']
+
             # 4. Sorting
             if sort is not None:
                 start = time.time()
                 if guid_list is True:
-                    data_list = [object_type(guid) for guid in data_list]
```

Comment:
Instantiating a new DataList results in the `._executed` to be `False`
The old notation resulted in an actual list with fully serialized objects, does the new notation have the same result?

Reply:
Yes, although serialization is still done when 'contents' is requested.
Now instead of manually creating a serialized list, everything is offloaded to DataList